src/Strategy/Composer.hs

```diff
@@ -99,6 +115,46 @@ instance FromJSON CompDep where
       <*> obj .:? "require"
       <*> obj .:? "require-dev"
 
+newtype ComposerJson = ComposerJson
+  { license :: Maybe [License]
```

Comment:
Part of what's made this parser so complicated is that we're trying to combine the `composer.json` schema with our internal data model.  We should separate these, and simply model the `composer.json` schema as-is.

Reply:
My instinct here is to write something like this:

```
newtype ComposerJson = ComposerJson { composerLicenses :: [Text] }

instance FromJSON ComposerJson where
  parseJSON = "much simpler parser which converts strings to singleton lists and missing keys to empty lists"

analyzeLicenses = do
  composerJson <- readContentsJson foo
  pure . map (licenseToResult . textToLicense) . filter Text.isEmpty $ composerLicenses composerJson 
```